text/src/autogluon/text/automm/data/collator.py

```diff
@@ -0,0 +1,324 @@
+            valid_length (NDArray, optional):
+                The sequences' original lengths at the padded axis. Shape is (N,). This will only be
+                returned if `ret_length` is True.
+
+        """
+        if isinstance(data[0], (torch.Tensor, np.ndarray, list, tuple)):
+            padded_arr, original_length = _pad_arrs_to_max_length(
+                data,
+                pad_axis=self._axis,
+                pad_val=self._pad_val,
+                round_to=self._round_to,
+                max_length=self._max_length,
+            )
+            if self._ret_length:
+                return padded_arr, original_length
+            else:
+                return padded_arr
+        else:
+            raise NotImplementedError
+
+
+class Tuple:
+    r"""Wrap multiple data collator functions together. The input functions will be applied
+    to the corresponding input fields.
+
+    Each data sample should be a list or tuple containing multiple attributes. The `i`th collate
+    function stored in `Tuple` will be applied on the `i`th attribute. For example, each
+    data sample is (nd_data, label). You can wrap two collate functions using
+    `Tuple(DataCollate, LabelCollate)` to collate nd_data and label correspondingly.
+
+    Arguments:
+        fn (list or tuple or callable): The collate functions to wrap.
+        *args (tuple of callable, optional): The additional collate functions to wrap.
+
+    """
+
+    def __init__(self, fn, *args):
+        if isinstance(fn, (list, tuple)):
+            if len(args) != 0:
+                raise ValueError(
+                    "Input pattern not understood. "
+                    "The input of Tuple can be Tuple(A, B, C) "
+                    "or Tuple([A, B, C]) or Tuple((A, B, C)). "
+                    "Received fn={}, args={}".format(str(fn), str(args))
+                )
+            self._fn = fn
+        else:
+            self._fn = (fn,) + args
+        for i, ele_fn in enumerate(self._fn):
+            if not hasattr(ele_fn, "__call__"):
+                raise ValueError(
+                    "Collate functions must be callable! "
+                    "type(fn[{}])={}".format(i, str(type(ele_fn)))
+                )
+
+    def __call__(self, data):
+        """Collate the input data.
+
+        Arguments:
+            data (list): The samples to collate. Each sample should contain N attributes.
+
+        Returns:
+            ret (tuple):
+                A tuple of length N. Contains the collated result of each attribute in the input.
+        """
+        if len(data[0]) != len(self._fn):
+            raise ValueError(
+                "The number of attributes in each data sample "
+                "should contains {} elements".format(len(self._fn))
+            )
+        ret = []
+        for i, ele_fn in enumerate(self._fn):
+            ret.append(ele_fn([ele[i] for ele in data]))
+        return tuple(ret)
+
+
+class List:
```

Comment:
It looks like the List is not used anywhere

Reply:
Just in case we may need it in future, I have added a unit test case for it too. Maybe we can keep it for now.